wdl-engine/src/inputs.rs

```diff
@@ -616,18 +646,50 @@ impl Inputs {
         let reader = BufReader::new(file);
         let object = mem::take(
             serde_json::from_reader::<_, JsonValue>(reader)
-                .with_context(|| {
-                    format!("failed to parse input file `{path}`", path = path.display())
-                })?
+                .context("failed to parse input file")?
                 .as_object_mut()
-                .with_context(|| {
-                    format!(
-                        "expected input file `{path}` to contain a JSON object",
-                        path = path.display()
-                    )
-                })?,
+                .context("expected input file to contain a JSON object")?,
         );
 
+        Self::parse_object(document, object).context("failed to parse input file")
+    }
+
+    /// Parses a YAML inputs file from the given file path.
+    ///
+    /// The parse uses the provided document to validate the input keys within
+    /// the file.
+    ///
+    /// Returns `Ok(Some(_))` if the file is a non-empty inputs.
+    ///
+    /// Returns `Ok(None)` if the file contains an empty input.
+    pub fn parse_yaml(
+        document: &Document,
+        path: impl AsRef<Path>,
+    ) -> Result<Option<(String, Self)>> {
+        let path = path.as_ref();
+
+        let file = File::open(path).with_context(|| {
+            format!("failed to open input file `{path}`", path = path.display())
+        })?;
+
+        // Parse the YAML
+        let reader = BufReader::new(file);
+        let yaml: YamlValue = serde_yaml_ng::from_reader(reader).with_context(|| {
+            format!("failed to parse input file `{path}`", path = path.display())
+        })?;
+
+        // Convert YAML to JSON format
+        let mut json =
+            serde_json::to_value(yaml).context("failed to convert YAML to JSON for processing")?;
```

Comment:
Sorry about this, as I think this is the line I told you to change to use `context` in the first place, but I previously failed to notice that this doesn't include the file path either. I think we should change it back to `with_context`, but include the path.

```suggestion
            serde_json::to_value(yaml).with_context(|| format!("failed to convert input file `{path}` to JSON for processing", path = path.display()))?;
```